go.mod

```diff
@@ -3,7 +3,11 @@ module github.com/p4lang/p4runtime
 go 1.14
 
 require (
-	github.com/golang/protobuf v1.4.0
-	google.golang.org/genproto v0.0.0-20200413115906-b5235f65be36
+	github.com/golang/protobuf v1.4.3
+	github.com/google/go-cmp v0.5.1 // indirect
+	golang.org/x/sys v0.0.0-20200930185726-fdedc70b468f // indirect
+	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
+	google.golang.org/genproto v0.0.0-20200526211855-cb27e3aa2013
 	google.golang.org/grpc v1.28.1
+	google.golang.org/protobuf v1.25.0 // indirect
```

Comment:
I am also unsure about the go.mod changes. I tried a few things (e.g., removing all required modules and running `go mod tidy`) but I could never get these indirect dependencies. Could you explain how they got there?

Reply:
I ran the codegen/update.sh. Without that the build was failing because the Go modules were not up to date.

Reply:
Also some of these might be due to Google auto updater which is defaulted in my emacs. Is there an issue with keeping all of these dependencies up-to-date? I can always try to handcraft the go.mod to remove the indirect ones and see if it is still ok with the build but I am a bit afraid messing with this by hand.